batch7crud-roll7/src/main/java/com/lftechnology/batch7crud/controller/StudentListController.java

```diff
@@ -0,0 +1,161 @@
+    try {
+      if (urlPath.length == 2 && CommonConstants.LIST_URL.equals(urlPath[1]))
+        list(request, response);
+      else if (urlPath.length == 3 && CommonConstants.NEW_ENTRY.equals(urlPath[2]))
+        create(request, response);
+      else if (urlPath.length == 4 && CommonConstants.EDIT.equals(urlPath[3]))
+        edit(request, response, Integer.parseInt(urlPath[2]));
+      else if (urlPath.length == 3)
+        show(request, response);
+      else{
+        showErrorPage(request, response);
+      }
+    } catch (DataException | ServletException | IOException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      showInternalErrorPage(request, response);
+    }
+  }
+
+  private void list(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException, DataException {
+    int pageSize = CommonConstants.PAGE_SIZE;
+    int page = 1;
+    try {
+      page = getPageNumber(request);
+    } catch (NumberFormatException e) {
+      showErrorPage(request, response);
+    }
+    int totalStudents = studentService.fetchTotal();
+    int numberOfPages = (int) Math.ceil(totalStudents / (float) pageSize);
+
+    if (page != 1 && page > numberOfPages ) {
+      page = 1;
+    }
+    request.setAttribute("students", studentService.fetch(pageSize, (page - 1) * pageSize));
+    request.setAttribute("pageSize", pageSize);
+    request.setAttribute("totalStudents", totalStudents);
+    request.setAttribute("pageNum", page);
+    request.setAttribute("numberOfPages",numberOfPages);
+    request.getServletContext().getRequestDispatcher(CommonConstants.STUDENTS_LIST_VIEW).forward(request, response);
+  }
+
+  private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    request.getServletContext().getRequestDispatcher(CommonConstants.NEW_ENTRY_VIEW).forward(request, response);
+  }
+
+  private void createProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException, DataException {
+    String name = request.getParameter("name");
+    String address = request.getParameter("address");
+    String roll = request.getParameter("roll");
+    try {
+      int rollNum = Integer.parseInt(roll);
+      Student student = new Student();
+      student.setName(name);
+      student.setAddress(address);
+      student.setRoll(rollNum);
+
+      studentService.addNew(student);
+      response.sendRedirect(request.getContextPath() + File.separator + CommonConstants.LIST_URL);
+    } catch (NumberFormatException e) {
+      request.setAttribute("error", "invalid roll");
+      request.getServletContext().getRequestDispatcher(CommonConstants.NEW_ENTRY_VIEW).forward(request, response);
+    }
+
+  }
+
+  private void edit(HttpServletRequest request, HttpServletResponse response, int id) throws ServletException, IOException, DataException {
+    request.setAttribute("student", studentService.fetchById(id));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "student" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)